tools/validators/instance_validator/validate/handler.py

```diff
@@ -0,0 +1,176 @@
+  for key, value in t.points.items():
+    print()
+    print('-point: ', key)
+    print('-- point_name: ', value.point_name)
+    print('-- present_value: ', value.present_value)
+  message.ack()
+
+
+def deserialize(yaml_file, universe):
+  """Parses a yaml configuration file and deserialize it.
+  Args:
+    yaml_file: the building configuration file.
+    universe: the generated ontology universe.
+  """
+  raw_parse = instance_parser.parse_yaml(yaml_file)
+  print('Passed syntax checks!')
+  print('Serializing Passed syntax checks!')
+  parsed = dict(raw_parse)
+  entity_instances = {}
+  entity_names = list(parsed.keys())
+  # first build all the entity instances
+  for entity_name in entity_names:
+    entity = dict(parsed[entity_name])
+    instance = entity_instance.EntityInstance(entity,
+                                              universe,
+                                              set(entity_names))
+    entity_instances[entity_name] = instance
+  return entity_instances
+
+
+class ValidatorHandler(object):
+  """A validator handler to coordiante the various steps of the validation."""
+
+  def __init__(self):
+    super().__init__()
+    self._PrepareArgs()
+
+  def Validate(self):
+    universe = self.GenerateUniverse(self.modified_types_filepath)
+    entity_instances = deserialize(self.filename, universe)
+    self.ValidateEntities(entity_instances)
+    self.TelemetryValidation(self.subscription, self.service_account,
+                             message_handler)
+
+  @staticmethod
+  def GenerateUniverse(modified_types_filepath=None):
+    """Generates the universe from the ontology.
+
+    Args:
+     modified_types_filepath: the path to a modified ontology.
+       If it is not set, the universe is generated from the default path.
+    """
+    # SYNTAX VALIDATION
+    print('\nValidator starting ...\n')
+    # prints for syntax errors and exits gracefully
+
+    print('Generating universe ...')
+    universe = generate_universe.BuildUniverse(modified_types_filepath)
+    if universe is None:
+      print('\nError generating universe')
+      sys.exit(0)
+
+    print('Universe generated successfully')
+    return universe
+
+  @staticmethod
+  def ValidateEntities(entity_instances):
+    """Validates entity instances that are already deserialized.
+      Args:
+        entity_instances: a list of entity instances.
+    """
+    print('Validating entities ...')
+    building_found = False
+    for entity_name, current_entity_instance in entity_instances.items():
+      if not current_entity_instance.IsValidEntityInstance(entity_instances):
+        print(entity_name, 'is not a valid instance')
+        sys.exit(0)
+      if current_entity_instance.type_name.lower() == 'building':
+        building_found = True
+
+    if not building_found:
+      print('Building Config must contain on entity with a building type')
+      sys.exit(0)
+    print('Entities Validated !')
+
+
+  def TelemetryValidation(self, subscription,
```

Comment:
This could be named more declaratively, ex: StartTelemetryValidation()

Reply:
done

Reply:
done